[bugfix](load) bitshuffle_page return ok when remaining element not e…

yixiutt opened the pull request:
…nough to enable column writer write new page

# Proposed changes

Issue Number: close #xxx

## Problem Summary:

page write return error, colomn writer will not change to next_page， be will core when flush mem_table cause this column data is null

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

dataroaring reviewed (approved):
LGTM

dataroaring reviewed (changes requested):
Please describe the problem and which commit introduced the problem.

Commit:
[bugfix](load) fix be crash in write ordinal_index  when flush mem_table

commit #9123 makes bitshuffle_page return error when remain capacity
is not enough, which lead to column_write return error and cannot
finish_current_page even when page is full, then be crash when flush
memtable, found that some column is empty, and crashed in build oridinal
index page.

dataroaring reviewed (dismissed):
LGTM
